docker-entrypoint/docker-entrypoint.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+	//For testing purposes
+	//"k8s.io/kubernetes/pkg/client/restclient"
+
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+const (
+	WAITING  = "waiting"
+	READY    = "ready"
+	RUNNING  = "Running"
+	INTERVAL = 2
+)
+
+var (
+	//"Info logger""
+	Info *log.Logger
+	//"Error logger"
+	Error *log.Logger
+)
+
+type dependency interface {
+	Exists(namespace string, name string) error
+	DepResolved(namespace string, name string) (bool, error)
+	GetType() string
+}
+
+func WaitFor(dep dependency, namespace string, names []string) error {
+	depState := WAITING
+	if names == nil {
+		depState = READY
+		Info.Println("Container has no ", dep.GetType(), " dependencies")
+	}
+
+	for depState == WAITING {
+
+		for _, name := range names {
+
+			name := strings.TrimSpace(name)
+			err := dep.Exists(namespace, name)
+			if err != nil {
+				Info.Println(dep.GetType(), name, " doesn't exists -> State waiting")
+				depState = WAITING
+				break
+			}
+			res, err := dep.DepResolved(namespace, name)
+			if err != nil {
+				Info.Println(dep.GetType(), name, " doesn't exists -> State waiting")
+				depState = WAITING
+				break
+			}
+			if !res {
+				Info.Println(dep.GetType(), name, " is not ready -> State waiting")
+				depState = WAITING
+				break
+			}
+
+			Info.Println(dep.GetType(), name, " is ready -> State ready")
+			depState = READY
+
+		}
+		//Sleep before next request
+		time.Sleep(INTERVAL * time.Second)
+	}
+	Info.Println("All", dep.GetType(), "dependencies resolved")
+	return nil
+}
+
+// This function executes command passed as array of strings
+func ExecuteCommand(command []string) error {
+	path, err := exec.LookPath(command[0])
+	if err != nil {
+		return err
+	}
+	cmd := exec.Cmd{
+		Path:   path,
+		Args:   command,
+		Stdout: os.Stdout,
+		Stderr: os.Stderr,
+	}
+	Info.Println("Executing command: ", command)
+
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func ConvertEnvToList(env string, s ...string) (out []string) {
+	sep := ","
+	if len(s) > 0 {
+		sep = s[0]
+	}
+	environ := os.Getenv(env)
+	if environ == "" {
+		return nil
+	}
+	out = strings.Split(environ, sep)
+	return out
+}
+
+func InitLogger(linfo io.Writer, lerror io.Writer) {
+	Info = log.New(linfo, "Entrypoint INFO: ", log.Ldate|log.Ltime|log.Lshortfile)
+	Error = log.New(lerror, "Entrypoint Error: ", log.Ldate|log.Ltime|log.Lshortfile)
+}
+
+func main() {
+
```

Comment:
Please run a go linter (i.e. https://github.com/alecthomas/gometalinter) on the code.